Cabal/Distribution/Simple/Program/HcPkg.hs

```diff
@@ -268,6 +278,53 @@ list hpi verbosity packagedb = do
   where
     parsePackageIds = sequence . map simpleParse . words
 
+-- Create a view from either name or symlink via filepath.
+createView :: HcPkgInfo -> Verbosity -> Either String FilePath -> IO ()
```

Comment:
Maybe views should be represented by an opaque data type, rather than as strings? Then, there is no problem for library users if we need to change the representation later.